src/tv2-common/content/dve.ts

```diff
@@ -483,6 +441,66 @@ export function MakeContentDVE2<
 	}
 }
 
+const setBoxSource = (
+	boxConfig: BoxConfig,
+	boxSources: BoxSources,
+	sourceInfo: { port: number; sourceLayerType: SourceLayerType }
+) => {
+	if (boxConfig) {
+		boxConfig.source = sourceInfo.port
+
+		boxSources.push({
+			// TODO - draw box geometry
+			...boxSource(sourceInfo),
+			...literal<CameraContent | RemoteContent>({
+				studioLabel: '',
+				switcherInput: sourceInfo.port
+			})
+		})
+	}
+}
+
+const setBoxToBlack = (boxConfig: BoxConfig, boxSources: BoxSources) => {
+	setBoxSource(boxConfig, boxSources, {
+		port: AtemSourceIndex.Blk,
+		sourceLayerType: SourceLayerType.UNKNOWN
+	})
+}
+
+function makeBoxMap(
+	inputs: string[],
+	context: IShowStyleUserContext,
+	classes: string[],
+	dveGeneratorOptions: DVEOptions,
+	sources: DVESources | undefined,
+	boxMap: Array<SourceDefinition | undefined>
+) {
+	inputs.forEach(source => {
+		const sourceProps = source.split(':')
+		const fromCue = sourceProps[1]
+		const targetBox = Number(sourceProps[0])
+		if (!fromCue || !targetBox || isNaN(targetBox)) {
+			context.notifyUserWarning(`Invalid DVE mapping: ${sourceProps}`)
+			return
+		}
+
+		classes.push(`${fromCue.replace(/\s/g, '')}_${dveGeneratorOptions.boxMappings[targetBox - 1]}`)
+
+		if (sources) {
```

Comment:
Since you moved this code it would be a good idea to boyscout the nesting away :D 
It could look like this:
```
const index: number = targetBox -1;

if (!sources) {
  boxMap[index] = undefined
  return;
}

const prop = sources[fromCue as keyof DVESources]
if (prop) {
  boxMap[index] = prop
  return;
}
context.notifyUserWarning(...);
boxMap[index] = undefined

```